app/controllers/github_repos_controller.rb

```diff
@@ -5,77 +5,65 @@ class GithubReposController < ApplicationController
   def index
     authorize GithubRepo
 
-    client = create_octokit_client
-
-    existing_user_repos = current_user.github_repos.where(featured: true).
-      distinct.pluck(:github_id_code)
+    known_repositories_ids = current_user.github_repos.featured.distinct.pluck(:github_id_code)
 
-    @repos = client.repositories.map do |repo|
-      repo.selected = existing_user_repos.include?(repo.id)
-      repo
-    end
+    # NOTE: this will invoke autopaging, by issuing multiple calls to GitHub
+    # to fetch all of the user's repositories. This could eventually become slow
+    @repos = fetch_repositories_from_github(known_repositories_ids)
   rescue Octokit::Unauthorized => e
-    render json: { error: "Github Unauthorized: #{e.message}", status: 401 }, status: :unauthorized
-  end
-
-  def create
-    authorize GithubRepo
-    @repo = GithubRepo.find_or_create(fetched_repo_params(fetch_repo))
-    current_user.touch(:github_repos_updated_at)
-    if @repo.valid?
-      redirect_to "/settings/integrations", notice: "GitHub repo added"
-    else
-      redirect_to "/settings/integrations",
-                  error: "There was an error adding your Github repo"
-    end
-  end
-
-  def update
```

Comment:
This comes from the previous move from the API, see #6304
I couldn't find any code that uses this, so I removed it.